addons/http_routing/models/ir_http.py

```diff
@@ -659,7 +659,7 @@ def url_rewrite(self, path):
         except werkzeug.exceptions.MethodNotAllowed:
             _ = router.match(path, method='GET')
         except werkzeug.routing.RequestRedirect as e:
-            new_url = e.new_url[7:]  # remove scheme
+            new_url = e.new_url.split('?')[0][7:]  # remove scheme, query string
```

Comment:
Might be a good idea to "properly" parse and process the URL? Because this is getting *really* obscure, and I feel something like
```suggestion
            new_url = url_parse(e.new_url).replace(scheme='', query='').to_url()
```
would be clearer (might still need to strip out the leading `//`, or maybe *just* keep the path itself?).

The hardcoded 7 also seems a bit odd, `https://` is 8 and `http` alone is 4, does it work exclusively for `http://` and... breal `https://` if that's ever encountered?

Reply:
Or is it that for some reason we get `https://boutique` and stripping the first 7 characters leaves `/boutique` which is what we want after all?

Reply:
URL in my tests were always in the form `http:///boutique`

checking in werzeug, there is no information on current domain / https scheme from router when filling new_url:

![2021-12-21-161057_1171x691_scrot](https://user-images.githubusercontent.com/9977887/146953513-84ae5b19-3832-4552-94d8-b0ec2ef59808.png)

Reply:
Indeed, I tried under https and the `new_url` form the `RequestRedirect` exception seems to always be `http://<relative_path>`, something like `http:///boutique` as you said (notice the triple slash)